app/src/main/java/com/example/coronadiagnosticapp/ui/fragments/welcome/AutostartUtils.java

```diff
@@ -0,0 +1,104 @@
+package com.example.coronadiagnosticapp.ui.fragments.welcome;
+
+import android.app.AlertDialog;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.Settings;
+import android.widget.CompoundButton;
+
+import androidx.appcompat.widget.AppCompatCheckBox;
+
+import com.example.coronadiagnosticapp.BuildConfig;
+import com.example.coronadiagnosticapp.R;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class AutostartUtils {
+
+    public static void requestAutostartPermissions(Context context) {
+        final List<Intent> SETTINGS_INTENTS = Arrays.asList(
```

Comment:
Just confirming, this list is everything we need? Cause some providers are missing here

Reply:
Talked about it on whatsapp.


Those are the only providers I read that needs this permission.
Though I didn't test this on other devices than my - xiaomi, and pixal emolator. The pixal indeed didn't need any other permission.